src/prefect/_internal/compatibility/migration.py

```diff
@@ -30,14 +30,23 @@
     "prefect.client:get_client": "prefect.client.orchestration:get_client",
 }
 
+upgrade_guide_msg = "Refer to the upgrade guide for more information: https://docs.prefect.io/latest/guides/upgrade-guide-agents-to-workers/."
+
 REMOVED_IN_V3 = {
     "prefect.client.schemas.objects:MinimalDeploymentSchedule": "Use `prefect.client.schemas.actions.DeploymentScheduleCreate` instead.",
     "prefect.deployments.deployments:Deployment": "Use `flow.serve()`, `flow.deploy()`, or `prefect deploy` instead.",
     "prefect.deployments:Deployment": "Use `flow.serve()`, `flow.deploy()`, or `prefect deploy` instead.",
-    "prefect.filesystems:GCS": "Use `prefect_gcp` instead.",
-    "prefect.filesystems:Azure": "Use `prefect_azure` instead.",
-    "prefect.filesystems:S3": "Use `prefect_aws` instead.",
+    "prefect.filesystems:GCS": "Use `prefect_gcp.GcsBucket` instead.",
+    "prefect.filesystems:Azure": "Use `prefect_azure.AzureBlobStorageContainer` instead.",
+    "prefect.filesystems:S3": "Use `prefect_aws.S3Bucket` instead.",
+    "prefect.filesystems:GitHub": "Use `prefect_github.GitHubRepository` instead.",
     "prefect.engine:_out_of_process_pause": "Use `prefect.flow_runs.pause_flow_run` instead.",
+    "prefect.agent:PrefectAgent": "Use workers instead. " + upgrade_guide_msg,
+    "prefect.infrastructure:KubernetesJob": "Use `prefect_kubernetes.KubernetesJob` instead.",
```

Comment:
In most cases, I think users will want to use the Kubernetes workers to replace their `KubernetesJob` usage, since they are likely using it as infrastructure with an agent.

Reply:
👍 !

Reply:
Applied in [c79af8d](https://github.com/PrefectHQ/prefect/pull/14348/commits/c79af8dcf04e880d61133681202c80fd43724187)